src/Features/WetnessEffects.h

```diff
@@ -66,4 +93,32 @@ struct WetnessEffects : Feature
 	virtual void RestoreDefaultSettings();
 	float CalculateWeatherTransitionPercentage(float skyCurrentWeatherPct, float beginFade, bool fadeIn);
 	void CalculateWetness(RE::TESWeather* weather, RE::Sky* sky, float seconds, float& wetness, float& puddleWetness);
+
+	virtual void PostPostLoad() override { Hooks::Install(); }
+
+	struct Hooks
+	{
+		struct BSParticleShader_SetupGeometry
+		{
+			static void thunk(RE::BSShader* This, RE::BSRenderPass* Pass, uint32_t RenderFlags)
+			{
+				func(This, Pass, RenderFlags);
+
+				auto particleShaderProperty = (RE::BSParticleShaderProperty*)Pass->shaderProperty;
+				auto cube = (RE::BSParticleShaderCubeEmitter*)particleShaderProperty->particleEmitter;
```

Comment:
This is actually a `BSParticleShaderRainEmitter` which is a child of `BSParticleShaderCubeEmitter`. We may want to do more RE to get the right class but maybe it's fine.

Reply:
VR/SE seem ok for RE for the members we care about.